conjure/unionwriter.go

```diff
@@ -106,6 +108,18 @@ func writeUnionType(file *jen.Group, unionDef *types.UnionType, genAcceptFuncs b
 			jen.Err().Op("!=").Nil(),
 		).Block(jen.Return(jen.Err())),
 		jen.Op("*").Id(unionReceiverName).Op("=").Id("deser").Dot("toStruct").Call(),
+		jen.Switch(jen.Id(unionReceiverName).Dot("typ")).BlockFunc(func(cases *jen.Group) {
+			for _, fieldDef := range unionDef.Fields {
+				cases.Case(jen.Lit(fieldDef.Name)).BlockFunc(func(caseBody *jen.Group) {
+					if !fieldDef.Type.IsOptional() {
+						selector := jen.Id(unionReceiverName).Dot(transforms.PrivateFieldName(fieldDef.Name))
+						caseBody.If(selector.Op("==").Nil()).Block(
+							jen.Return(snip.FmtErrorf().Call(jen.Lit(fmt.Sprintf("field %s is required", fieldDef.Name)))),
```

Comment:
Looking at generated code, I wonder if error would be more readable if field name was quoted instead (basically have a `fmt.Sprintf` call with `%q` as formatter and `fieldDef.Name` as value)